File: bubble_sort/bubble_sort.go
// Package main provides ...
package main

import (
	"fmt"
)

func main() {
	list := []int{1, 3, 2, 8, 4, 10, 54, 49, 6}
	fmt.Println("unsorted: ", list)

	bubble_sort(list)
	fmt.Println("sorted: ", list)
}

func bubble_sort(a []int) {
	for itemCount := len(a) - 1; ; itemCount-- {
		swap := false
		for i := 1; i <= itemCount; i++ {
			if a[i-1] > a[i] {
				a[i-1], a[i] = a[i], a[i-1]
				swap = true
			}
		}
		if swap == false {
			break
		}
	}
}

File: bubble_sort/bubble_sort.js
function bubble_sort(arr) {
  var len = arr.length,
    swap = false;

  do {
    swap = false;
    for (var i = 1;  i <= len; i ++) {
      if (arr[i - 1] > arr[i]) {
        var temp = arr[i];
        arr[i] = arr[i - 1];
        arr[i - 1] = temp;
        swap = true;
      }
    }

    len--;
  } while (swap);

  return arr
}

var arr = [1, 6, 2, 3, 9 , 4, 11, 10, 22, 8];

console.log(bubble_sort(arr));
